src/UnitTests/ValueTransformers.cs

```diff
@@ -0,0 +1,247 @@
+﻿using Shouldly;
+using Xunit;
+
+namespace AutoMapper.UnitTests
+{
+    namespace ValueTransformers
+    {
+        public class BasicTransforming : AutoMapperSpecBase
+        {
+            public class Source
+            {
+                public string Value { get; set; }
+            }
+
+            public class Dest
+            {
+                public string Value { get; set; }
+            }
+
+            protected override MapperConfiguration Configuration { get; } = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>();
+                cfg.ApplyTransform<string>(dest => dest + " is straight up dope");
+            });
+
+            [Fact]
+            public void Should_transform_value()
+            {
+                var source = new Source
+                {
+                    Value = "Jimmy"
+                };
+                var dest = Mapper.Map<Source, Dest>(source);
+
+                dest.Value.ShouldBe("Jimmy is straight up dope");
+            }
+        }
+
+        public class StackingTransformers : AutoMapperSpecBase
+        {
+            public class Source
+            {
+                public string Value { get; set; }
+            }
+
+            public class Dest
+            {
+                public string Value { get; set; }
+            }
+
+            protected override MapperConfiguration Configuration { get; } = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>();
+                cfg.ApplyTransform<string>(dest => dest + " is straight up dope");
+                cfg.ApplyTransform<string>(dest => dest + "! No joke!");
```

Comment:
I've been following this at a little bit of a distance. FWIW, from a functionality standpoint I like the idea of stacking transforms, but it seems like a departure from the rest of the API, which I think just does overrides. On the whole I think I would vote for `ApplyTransform()` and overrides, rather than `AddTransform()` and stacking, but I don't feel strongly about it.